.github/workflows/test.yaml

```diff
@@ -50,75 +54,95 @@ jobs:
                   echo "The Git workspace is clean. No changes detected."
                 fi
 
-    tests-php-components:
-        runs-on: ubuntu-latest
-        outputs:
-            components: ${{ steps.components.outputs.components }}
-        steps:
-            - uses: actions/checkout@v4
-
-            - id: components
-              run: |
-                components=$(find src/ -mindepth 2 -type f -name composer.json -not -path "*/vendor/*" -printf '%h\n' | jq -R -s -c 'split("\n")[:-1] | map(. | sub("^src/";"")) | sort')
-                echo "$components"
-                echo "components=$components" >> $GITHUB_OUTPUT
-
     tests-php:
         runs-on: ubuntu-latest
-        needs: tests-php-components
         strategy:
             fail-fast: false
             matrix:
-                php-version: ['8.1', '8.3', '8.4']
-                minimum-stability: ['stable', 'dev']
-                include:
-                  - php-version: '8.1'
-                    dependency-version: 'lowest'
-                  - php-version: '8.3'
-                    dependency-version: 'highest'
-                  - php-version: '8.4'
-                    dependency-version: 'highest'
-                component: ${{ fromJson(needs.tests-php-components.outputs.components )}}
-                exclude:
-                  - php-version: '8.1'
-                    minimum-stability: 'dev'
-                  - php-version: '8.3'
-                    minimum-stability: 'dev'
-                  - component: Swup  # has no tests
-                  - component: Turbo # has its own workflow (test-turbo.yml)
-                  - component: Typed  # has no tests
-
+                config:
```

Comment:
instead of a single config key, I would still use separate matrix properties. It is much more common:

```
matrix:
  php-version: ['8.1', '8.2', '8.3', '8.4']
  composer-flags: ['']
  symfony-version: ['']
  minimum-stability: ['stable']
  include: # used to add additional jobs not belonging to default matrix
    # dev packages (probably not needed to have multiple such jobs)
    - minimum-stability: 'dev'
      php-version: '8.4'
    # lowest deps
    - php-version: '8.1'
      composer-flags: '--prefer-lowest'
    # LTS version of Symfony
    - php-version: '8.1'
      symfony-version: '6.4.*'
```

This will give you 7 jobs, and will favor jobs running unmodified dependency constraints (closer to what projects using a package will experience)

Reply:
That's a smart approach, the best of two worlds. Thanks :)